core/query.go

```diff
@@ -74,10 +70,6 @@ func NewRoot(ctx context.Context, opts QueryOpts) (*Query, error) {
 		return nil, fmt.Errorf("buildkit client: %w", err)
 	}
 
-	// NOTE: context.WithoutCancel is used because if the provided context is canceled, buildkit can
-	// leave internal progress contexts open and leak goroutines.
-	bk.WriteStatusesTo(context.WithoutCancel(ctx), opts.Recorder)
```

Comment:
Looking through the buildkit code, I'm a bit concerned that if we don't read its progress it will still end up writing it (including exec-op stdout/stderr) and then caching that all in memory for the duration of the job since it's never read.

I am not 100% sure that's actually what happens, but may be worth a test with a container that writes a ton to stdio and check whether the memory explodes. If it does, I guess we could leave in a reader for the progress that just discards it (for now anyways, obviously would be nice if we could modify upstream to just disable writing any of this)